Section 1: Introduction and Setup/4-RoR Development Environment/testing_project/app/channels/application_cable/connection.rb

```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
```

Comment:
what is this file for?

Reply:
`connection.rb` inherits from ActionCable::Connection::Base and its responsible for authentication and authorization of the connections from current users.

Reply:
This kind of answer shows me that you jstu copy paste what you read without understanding it. If you going to learn like this, then better not waste your time

How is it responsible for authentication and authorization? Where does it does that? How does it does that? Magically?

Whats the difference between authorization & authentication?

Reply:
Authentication using [Devise](https://github.com/heartcombo/devise)
Adding the code below in `connection.rb`:
```
module ApplicationCable
  class Connection < ActionCable::Connection::Base
    identified_by :current_user

    def connect
      self.current_user = find_verified_user
      logger.add_tags 'ActionCable', current_user.email
    end

    protected

    def find_verified_user # this checks whether a user is authenticated with devise
      if verified_user = env['warden'].user
        verified_user
      else
        reject_unauthorized_connection
      end
    end
  end
end

```
`authentication` is the process of verifying the identity of the consumers while `authorization` authorizes the consumers access to specific part of the sites.

Reply:
1. You are showing me EXTRA code that you ADD in yourself to do authentication by using the devise gem. This file is NOT responsible for doing it. Let me prove it to you. Move the entire code for authentication out of this file and put it in another file. Then, this file now does not have any authentication. So, I ask you again. What does this file have to do with being responsible for authentication & authorization?

Reply:
Rails by default creates a `ApplicationCable::Connection` class inside the `application_cable/connection.rb` file. For each WebSocket connection established, a connection object is created which becomes the parent for all channel subscriptions that are created after on.
>How is it responsible for authentication and authorization? Where does it does that? How does it does that? Magically?
I thought you mean by how to implement it inside the file, hence the code, then recent comment only I realized that you mean that it wasn't meant for authentication and authorization as default.

Reply:
> For each WebSocket connection established, a connection object is created which becomes the parent for all channel subscriptions that are created after on.

What object & how?

Reply:
Connection object is an instance of `ApplicationCable::Connection` which inherits `ActionCable::Connection::Base`.

On the client-side, consumers require an instance of the connection and a consumer becomes a subscriber by creating a subscription to a channel. 

Rust by default creates the following file `app/javascript/channels/consumer.js`
```
// app/javascript/channels/consumer.js

import { createConsumer } from "@rails/actioncable"

export default createConsumer()

```
To subscribe to a specific channel:
```
// app/javascript/channels/chat_channel.js

import consumer from "./consumer"

consumer.subscriptions.create({ channel: "ChatChannel" })
```